lib/mongo/server.rb

```diff
@@ -64,10 +64,18 @@ def initialize(address, cluster, monitoring, event_listeners, options = {})
       @connection_id_gen = Class.new do
         include Id
       end
-      @monitor = Monitor.new(address, event_listeners, monitoring,
-        options.merge(app_metadata: Monitor::AppMetadata.new(cluster.options)))
-      unless monitor == false
-        start_monitoring
+      @scan_semaphore = Semaphore.new
+      @round_trip_time_averager = RoundTripTimeAverager.new
+      @description = Description.new(address, {})
+      @last_scan = nil
+      unless options[:monitoring_io] == false
+        @monitor = Monitor.new(self, event_listeners, monitoring,
+          options.merge(
+            app_metadata: Monitor::AppMetadata.new(cluster.options),
+        ))
+        unless monitor == false
```

Comment:
`monitor` refers to the local variable here, not the instance method.

Reply:
Can we name it something else? I think it's fairly subtle that this isn't invoking the getter for `@monitor`.

Reply:
I agree that in principle a different name would be better, however given that this variable is assigned in the same method as well as the corresponding option being mentioned in the docstring, I think there is sufficient context provided to understand what is happening. If you would like to rename the variable in a separate PR I would be happy to review that.

Reply:
I think it's pretty confusing that there are three things called "monitor" here that don't all refer to the same thing, and only two of them are the same thing. I don't see why the local variable has to be called the same thing as the option, and I don't see why this shouldn't be done in this PR, since the confusing code is added in this PR.

Reply:
This PR does not add any new `monitor` references.

Reply:
I don't see why that's relevant? This PR adds new code that's hard to read due to ambiguity. Sometimes resolving that ambiguity means changing existing code.

Reply:
I think we now both understand what is happening, do you have any other comments regarding changes this PR is making?

Reply:
I understand what's happening now because of your explanation. Someone reading the code in the future won't have the benefit of that explanation and might not understand. I feel strongly that this code should be changed to not be confusing, since changing the name of the local variable is so easy and and clears things up significantly.